blocks/bricks/recurrent/base.py

```diff
@@ -46,6 +46,9 @@ def initial_states(self, batch_size, *args, **kwargs):
             The keyword arguments of the application call.
 
         """
+        if not hasattr(self, 'apply') or not self.apply.states:
+            return
+
```

Comment:
Why do you want to have a class without `apply`? It's a mistake if a user forgot to define `apply` and the best is to crash soon.

In a case if `apply.states` is empty, `initial_states` would return an empty list before this change, why is it wrong?